File: Jay/lv1/176963. 추억 점수 /추억 점수.swift
import Foundation

func solution(_ name:[String], _ yearning:[Int], _ photo:[[String]]) -> [Int] {
    var dic: [String: Int] = [:]
    var array = [Int]()
    for i in 0..<name.count {
        dic[name[i]] = yearning[i]
    }
    for i in photo {
        var result = 0
        for j in i {
            if let value = dic[j] {
                result += value
            }
        }
        array.append(result)
    }
    return array
}

File: Jay/lv1/176963. 추억 점수 /추억 점수.kt
class Solution {
    fun solution(name: Array<String>, yearning: IntArray, photo: Array<Array<String>>): IntArray {
    val dic = mutableMapOf<String, Int>()
    val array = mutableListOf<Int>()
    for (i in name.indices) {
        dic[name[i]] = yearning[i]
    }
    for (i in photo) {
        var result = 0
        for (j in i) {
            val value = dic[j]
            if (value != null) {
                result += value
            }
        }
        array.add(result)
    }
    return array.toIntArray()
    }
}
